com.unity.render-pipelines.high-definition/Runtime/Water/ShaderPassWaterForward.hlsl

```diff
@@ -0,0 +1,210 @@
+#if SHADERPASS != SHADERPASS_FORWARD
+#error SHADERPASS_is_not_correctly_define
+#endif
+
+// Given that the geometry is either procedural or a mesh given by the user, we only need this as an input
+struct VaryingsToPS
+{
+    VaryingsMeshToPS vmesh;
+};
+
+VaryingsMeshToPS VertMeshWater(AttributesMesh input)
+{
+    VaryingsMeshToPS output;
+    ZERO_INITIALIZE(VaryingsMeshToPS, output); // Only required with custom interpolator to quiet the shader compiler about not fully initialized struct
+
+    // Set up the instance data
+    UNITY_SETUP_INSTANCE_ID(input);
+    UNITY_TRANSFER_INSTANCE_ID(input, output);
+
+    // Apply the mesh modifications that come from the shader graph
+    float3 positionOS;
```

Comment:
note for me

Reply:
Note for self: evaluate the need of duplicated the shaderpasswaterforward.hlsl instead of reusing regular file